src/service/viewer-impl.js

```diff
@@ -835,6 +835,34 @@ export class Viewer {
   }
 
   /**
+   *
+   * @return {!Promise}
+   */
+  replaceShareTrackingFragment(incomingFragment, outgoingFragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        let newUrl = this.win.location.href;
+        if (incomingFragment == "") {
+          if (newUrl.indexOf('#') == -1) {
+            newUrl = newUrl + '#.' + outgoingFragment;
+          } else {
+            newUrl = newUrl.replace('#', '#.' + outgoingFragment + '&');
+          }
+        } else {
+          newUrl = newUrl.replace(incomingFragment, outgoingFragment);
+        }
+        this.win.history.replaceState({}, '', newUrl);
```

Comment:
That sounds the same to me for both cases :) My only concern that `replaceState` will break iframe navigation. Feel free to check that it doesn't and proceed with your solution, but the check is necessary.